### PR TITLE
fix(backend): 🍰 Fix unintentional exposure of reports

### PR DESCRIPTION
**Pull request by Tirokk**
## 🍰 Pullrequest

Hot fix: Set sensible report properties to moderators only in permissionsMiddleware.

### Issues

- relates #3074
- Original PR #3075

### Todo

None

### CODE REVIEW

**Comment by mattwr18** [comment]
hey @Tirokk ... I tried to add back the tests, but I also refactored to return the `resource` instead of needlessly exposing the `Report`, without just cause.
I hope I am not stepping on your toes. I know that you said you had another `PR` that you are working on, or one in mind... we have been dealing with `build` issues since last `Friday`, at least, so we couldn't deploy any fix, until we get it figured out. 
I guess we can talk about this approach and see if people like it, or have other ideas and go from there. 
Again, sorry if I haven't understood what you meant, maybe you could let me know if there is any way I can improve this. I have reread your comments in `Discord` and I am still lost, to be honest.

**Comment by mattwr18** [comment]
please let's discuss this in the daily tomorrow?

File: backend/src/schema/resolvers/reports.js [hunk 1]
```diff
@@ -38,12 +29,12 @@ export default {
...
-        return reportTransactionResponse.records.map(transformReturnType)
+        return reportTransactionResponse.records.map(record => record.get('returnedResource'))
       })
       try {
-        const [createdRelationshipWithNestedAttributes] = await reportWriteTxResultPromise
-        if (!createdRelationshipWithNestedAttributes) return null
-        return createdRelationshipWithNestedAttributes
+        const [reportedResource] = await reportWriteTxResultPromise
+        if (!reportedResource) return null
+        return reportedResource
```
Comment (roschaefer) [inline backend/src/schema/resolvers/reports.js]:
```suggestion
        return reportedResource || null
```
If you want to convert `undefined` to `null`.

Comment (mattwr18) [reply backend/src/schema/resolvers/reports.js]: that makes sense!

Comment (Tirokk) [reply backend/src/schema/resolvers/reports.js]: Yes, I like this better! 😍 @roschaefer

**Comment by Tirokk** [comment]
Will be solved as a whole in #3075

**Comment by Tirokk** [comment]
Will be solved as a whole in #3075

**Pull request closed without merging**